### PR TITLE
VSFP-113 Grammarly Part 4

### PR DESCRIPTION
**Pull request by fl3pp**

### CODE REVIEW

File: SA/design_concept/content/results/outlook_ide.typ [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
-An integration with e.g. the Haskell Substitution Stepper #footnote("https://eprints.ost.ch/id/eprint/991/") could be useful.
-The already-built type-inference engine could be replaced with a Haskell compiler run as well.
+Besides the functionality already in the #ac("PoC") application, additional
+features offered by Haskell would also need to be implemented visually, such as
+an editor for Haskell-type definitions, a way to organize user definitions into
+modules, integration with a package manager, etc.
+
+On the other hand, such a development would offer the opportunity to outsource
+functionality, such as compilation and execution, to Haskell tooling.
+An integration with, e.g., the Haskell Substitution Stepper #footnote("https://eprints.ost.ch/id/eprint/991/") could be helpful.
```
Comment (lstreckeisen) [inline SA/design_concept/content/results/outlook_ide.typ:15]: I'd switch the order of this and the next sentence

File: SA/design_concept/content/results/results.typ [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
 #include_section("design_concept/content/results/results_concept.typ", heading_increase: 1)
 #include_section("design_concept/content/results/results_poc.typ", heading_increase: 1)
 #include_section("design_concept/content/results/results_ui_demonstration.typ", heading_increase: 1)
 
-#include_section("design_concept/content/results/outlook.typ")
\ No newline at end of file
+= Outlook
+
+During the #ac("PoC") development, two possible variants of a feature-complete
+application emerged. Both are which are briefly described below.
```
Comment (lstreckeisen) [inline SA/design_concept/content/results/results.typ:21]:
```suggestion
application emerged. Both are briefly described below.
```

File: SA/design_concept/content/results/results_poc.typ [hunk 3]
```diff
@@ -1,24 +1,26 @@
 #import "../../../acronyms.typ": *
 
 = Proof of Concept
+
 The project produced a proof of concept application that implements the design concept described in @design-iteration-2.
-Although the application doesn't offer visualizations for the most commen aspects of functional programming, such as the construction of custom data types, the application shows that the proposed concept for function composition works and is easy to use.
-This means that the main goal of the project was achieved.
+Although the application doesn't offer visualizations for the most common aspects of functional programming, such as the construction of custom data types, the application shows that the proposed concept for function composition works and is easy to use.
```
Comment (lstreckeisen) [inline SA/design_concept/content/results/results_poc.typ:6]:
```suggestion
Although the application doesn't offer visualizations for some of the most common aspects of functional programming, such as the construction of custom data types, the application shows that the proposed concept for function composition works and is easy to use.
```

File: SA/design_concept/content/results/results_poc.typ [hunk 4]
```diff
@@ -1,24 +1,26 @@
...
+But a #ac("UI") technology that doesn't require a local web server, probably a GHCJS-based framework, would be better suited to implement a full version of VisualFP, as it would enable the application to be served to any browser as a set of static files.
 A different #ac("UI") framework may also provide better support for functional reactive programming, which is expected to make the #ac("UI") implementation more concise.
 
 == macOS Electron App
+
 For the project submission, the #ac("PoC") Electron application was packaged for Windows, Linux & macOS.
 Unfortunately, the macOS app has an unforeseen issue: It doesn't pass the macOS gatekeeper checks.
 
-Based on code signature, notarization and comparison with known malware, gatekeeper flags potentially dangerous applications and restricts them from execution.
+Based on code signature, notarization, and comparison with known malware, gatekeeper flag potentially dangerous applications and restrict them from execution.
```
Comment (lstreckeisen) [inline SA/design_concept/content/results/results_poc.typ:21]:
```suggestion
Based on code signature, notarization, and comparison with known malware, gatekeeper flags potentially dangerous applications and restricts them from execution.
```

File: SA/design_concept/content/results/results_ui_demonstration.typ [hunk 5]
```diff
@@ -1,27 +1,36 @@
 #import "../../../acronyms.typ": *
 
 = UI Demonstration
-@visual-fp-ui-demo-part-one and @visual-fp-ui-demo-part-two depict a step-by-step demonstration of the #ac("PoC") application
-using the `mapAdd5` code scenario as descriped in @design_eval_code_scenarios.
+
+@visual-fp-ui-demo-step-one, @visual-fp-ui-demo-step-two, @visual-fp-ui-demo-step-three, @visual-fp-ui-demo-step-four, @visual-fp-ui-demo-step-five and @visual-fp-ui-demo-step-six depict a step-by-step demonstration of the #ac("PoC") application
```
Comment (lstreckeisen) [inline SA/design_concept/content/results/results_ui_demonstration.typ:5]: I think something like `Figure X through Figure Y depcit ...` would be easier to read.

File: SA/design_concept/content/results/results_ui_demonstration.typ [hunk 6]
```diff
@@ -1,27 +1,36 @@
...
+using the `mapAdd5` code scenario as described in @design_eval_code_scenarios.
```
Comment (lstreckeisen) [inline SA/design_concept/content/results/results_ui_demonstration.typ:6]:
```suggestion
using the `mapAdd5` code scenario described in @design_eval_code_scenarios.
```

File: SA/design_concept/content/results/outlook_ide.typ (hunk 1 above)
Comment (fl3pp) [reply SA/design_concept/content/results/outlook_ide.typ]: Updated

File: SA/design_concept/content/results/results_ui_demonstration.typ (hunk 5 above)
Comment (fl3pp) [reply SA/design_concept/content/results/results_ui_demonstration.typ]: Updated

**Review by lstreckeisen — APPROVED** [review]

**Pull request merged**